### DISCUSSION TITLE
Support for pyvis and networkx

### DISCUSSION THREAD

**Discussion by anirbanbasu** [widgetti/solara#632] [Q&A] [answered]
Hi, is there any planned support for `pyvis` and `networkx` to display graphs? The HTML that `pyvis.network.Network` generates does not render with the `solara.HTML` component.

**Comment by maartenbreddels** [comment, marked as answer]
Hi,

if it displays in Jupyter notebooks, you can try:
https://solara.dev/documentation/api/utilities/display

Regards,

Maarten

**Reply by anirbanbasu** [reply]
Thanks @maartenbreddels. I haven't tried in Jupyter but reading about it on the Internet, it seems like the `show` method on the `pyvis.network.Network` can be used to display the graph in Jupyter. However, this creates a static HTML file and stores it on the disk, which is a no go for my use-case.

However, after all, it appears to display that HTML file in an iframe. Thus, I tried `solara.display()` with the HTML generated by the `generate_html()` method on `pyvis.network.Network`. All I got is the entire static HTML source with the necessary Javascript, not the rendered graph.

I had also tried displaying the static HTML using an iframe to no avail. See the code https://github.com/anirbanbasu/tldrlc/blob/61f76e7a291fc15bc6a95885531de586cfb34ac0/pages/kg.py#L156

**Reply by EwoutH** [reply]
Just joining in that I would love networkx graph visualisation in Solara!

**Reply by maartenbreddels** [reply]
Getting back to the iframe issue, I don't understand why it's not working, in general it works:

https://py.cafe/snippet/solara/v1#c=H4sIAKApjWYAA3VX3W7jNhN9FUGL78sWiGVL9jqOt-oPigK9aIGiKHpTFwUlUhbXEsklqcTaIO_eM6TkeNOtDSThkJyfMzNnmKe01lyk-_RNUreiPiV68EnrvXH75dLpjlmWcfGwTBptE67roRfKMy-1Oqg3CWTz2aP07VBlte6Xj5Ifhfdyur-kk98r_gUDZsxq1ohlz5j1QlVWcC46F4z12opEnFlvOuEOSvZGW59ElQdYh6ixrBd_UwBJmRzS8FVft77vvsHvSvORfsdjibN1eUhfWxZ9Jfhr-8sH-cnqBRe9_ta3ohdlJ4-t_38n1el3_b0xZcM6Jw5pglB9C7X5avU_LFtB517Wzo-dwLLSlgu7T3JzpggkT96ILX3fJ3FrYRmXg9snO3N-f0i_-XoZvSb_l3MgyymyOVL6fjflCLgbrZCbg-KiSX5lR_H2q_1BJfg8IjUTcNkPuht69TY69nRIuXSmY-Mh3QPAphPnQ3qLv2IgURpjeZ610WdS9tPvv_z81rMjIozu0mXmvZXV4IUj_UAdVUOKrrJFFgJy1wa-bJakwdkXDwEjuUPRf8mfFyl9BuWQ5r-lUsLSdtkc0rkiamfKm3hgIRWSK25CldywCkW6rzqmTpAEqG56Zo9SLSrtve73CRu8fp9MQq_NLAlhwUM4_34qh3mlH4RtOv04H52LYhWsAqXy5ukKpOebqbpu6PrNXFtxdVUg6e3nEb_C_0vYvVz4Kr1Nrfg4SIsqV96BCOYOw44fDVFDlGDNjPlDisd0H8r_NhVc-h8Vqzqc8naAxIy-1Qp3zKi55GLxsMqKd1mOy6gyoJrun1Krtf9NQ83TbMFidZvWrey4Fbj-52XHs8oJ_9-b2DHMt1jAucyM6fNf-D7fpo0EbeDwX7Cs6xMtf9AKTQ4XQHYtc0gTX213K1Hc59WmyBtWbSuer3f5arPZFbt3d81BMTVKXZabbJOtsLJHrYpF3TSyLIt1lr8SLiqpuFRHh908K7JV8gY8Vp_oEGIsyzxbhysOVXQSis5Bcw4JskarSWfF6lPNuq4sgV8QCBSNbIbO6cFsyKG8yArIO8Hqtiy32YqO1VgIr3UHXe9gC5prYXGR_F1B-11W4FYMIM9ypAarllmAvFDa9qyTn4QtyzUu08FO1qey3MGrO6x030ePsMVFNRzNSGqwfQmUi1pb5jV0QDmpBx8NTvBzH6K5I5E4ixrhqCPFvyJJA0g-OK0c_O8ZiYsQEOWN8kcewdsNJB-5IqPB9TbPSWkestMOPVNwP6ATbkuuoGtN16QZT8IqASe2VJKIACKq1hDelrRNgsVRKHhHGE7oY-NhgF7EE0KKa8Aawr-f9Me5h1sE1xoip_kALDAsCX26C10f0DXB5_uwkuoDK0J05BJBYLREnYYchCCuYUFgd6T6wwBXMTaQH1Q0WURQV2LMTsIf2X4RigfqcDK9vVbhhAUxEeKE2SWPn28v8GcvFdqe7qOyLgc6Vi3MeAzsQXuhvq_2LqBQNDmwA2eeuH4EJOts-7JcSA89l6BJPBii5uAZGeyZN532nawmtg4oUl32gDkUV8Cwl84PtEuqsFZV3THnZB3ydxUibUT4cDNYhUQrhAsR6nRLeVUVHiMwHcp5FSTC-QVzo6qJGXKgSTe1F5XWqNMt4LmyMckXrpVT75CKoY-dUwQbNBks2BIo3SFnUGeYwkBA8QNFSKdqN-hTmFyh4YCHEWcjaji2QQW-WDToWFAfejoAdJe9I1mHwQM9iDKo75insLgk7ZvY0MbqXuC5M7hLWcHdeB5bxmPO6e4kISdkNwDeOOoTguY-2714AHq2uhYuFESMZ7ACBcgmQiNrY03hxMorKLhLESHfoR3N2IfKEGfQtkMXERIUfNiL84Xec1FnEFqH5M8pvQ8oxZbmzIvoawEFwX6QU3MtOn08RkfgLm2NjMiKODVo_dR_xCY45gpmC9711NhEf7BDUxSFEdxfh0a3Tb1er-8XiFrCPDEiVSjyTTv3u-3rHdiyksg8J56eVpQLor_ZjOEu9Aku7K7ccULxwltMtctRUPpgplmwJUpC1E6eQxeEmnV4QTexhsOkiHM-LMNMiusLQbwSD2GCzCL0qgTBXOaZ85hghDr0TYQDke3wf0Eoy3Xg1YlVeChqxIOLmApj7RwocZqVgCeeCK2CEKWHljDfQPvQS48BwuRfeUZXoH6LzWqdg2hwDAPns2LaRJgHK8Fv-D8Ddy9mwShgmnVkH1o_SLAq8g3fQ1k8Ml-3XGOEfV4YQR5eH6EurwnnsY4vulivcP1RgHA6HSb_JhIJRAKDdnpFENeSLVE5aMCMnjuTchxPxw1qjQDpxLd4MMyBRgeJep3r8ONjJ72A8fT5H_IR1r79DQAA


Is this a pyvis specific problem?
